sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -241,30 +244,115 @@ object Value {
         case (ContractId.V1(hash1, suffix1), ContractId.V1(hash2, suffix2)) =>
           hash1 ?|? hash2 |+| suffix1 ?|? suffix2
       }
+    }
+
+    final case class V2 private (local: Bytes, suffix: Bytes) extends ContractId with data.NoCopy {
+      override lazy val toBytes: Bytes = V2.prefix ++ local ++ suffix
+      lazy val coid: Ref.HexString = toBytes.toHexString
+      override def toString: String = s"ContractId($coid)"
+    }
+
+    object V2 {
+      private[lf] def MaxSuffixLength: Int = V1.MaxSuffixLength
```

Comment:
If I remember correctly `V1.MaxSuffixLength`, was such that a serialized cid was less or equal than 128 bytes:
(1 bytes for the length) + (1 byte for the version) + (32 bytes for the descriminator) + (max 94 bytes for the suffix)  

Not sure it make  lot of sens to reuse the same value for V2.

Reply:
I let you judge what is the best limit. 
Since the new scheme is made to be short, it makes sens for me to have a "short" suffix.

Reply:
Yep. I've set it now to 33 bytes.